crates/core/c8y_api/src/utils.rs

```diff
@@ -4,26 +4,28 @@ pub mod bridge {
     use tedge_api::MQTT_BRIDGE_DOWN_PAYLOAD;
     use tedge_api::MQTT_BRIDGE_UP_PAYLOAD;
 
-    pub fn is_c8y_bridge_up(message: &Message, service: &str) -> bool {
-        let c8y_bridge_health_topic = main_device_health_topic(service);
-        match message.payload_str() {
-            Ok(payload) => {
-                message.topic.name == c8y_bridge_health_topic && payload == MQTT_BRIDGE_UP_PAYLOAD
-            }
-            Err(_err) => false,
-        }
-    }
-
     pub fn is_c8y_bridge_established(message: &Message, service: &str) -> bool {
         let c8y_bridge_health_topic = main_device_health_topic(service);
         match message.payload_str() {
             Ok(payload) => {
                 message.topic.name == c8y_bridge_health_topic
-                    && (payload == MQTT_BRIDGE_UP_PAYLOAD || payload == MQTT_BRIDGE_DOWN_PAYLOAD)
+                    && (payload == MQTT_BRIDGE_UP_PAYLOAD
+                        || payload == MQTT_BRIDGE_DOWN_PAYLOAD
+                        || is_valid_status_payload(payload))
             }
             Err(_err) => false,
         }
     }
+
+    #[derive(serde::Deserialize)]
+    struct HealthStatus<'a> {
+        status: &'a str,
+    }
+
+    fn is_valid_status_payload(payload: &str) -> bool {
+        serde_json::from_str::<HealthStatus>(payload)
+            .map_or(false, |h| h.status == "up" || h.status == "down")
+    }
```

Comment:
[minor] I would move into `tedge_api` this logic to accept `0` | `1` | `status = up` | `status = down`.

Reply:
If I'm removing the check for bridge health, then this more complicated logic isn't required, so unless we need it for some other reason you know of, I'll undo this change

Reply:
Checking that the bridge has been initialized before sending registration messages to c8y is required for a very new thin-edge device. If the c8y mapper starts before the bridge and even before the c8y.url is provided (as this is the case for a generic image of thin-edge), then the mapper must wait for the bridge to be initialized - otherwise all the registration messages are lost.

Reply:
So this logic is required - but can be simplified by simply checking that at least one message has been received on this topic - whatever the payload.

Reply:
I just saw https://github.com/thin-edge/thin-edge.io/pull/2810/commits/4ef0fcd5aa235612f682fb725cd8c031073ba4a6 and I'm okay with it.